---
title: "Cotton fields soil microbiome Project"
author: "Matthew Gillis and Pedro Rodrigues"
output:
  BiocStyle::html_document:
    toc: true
    toc_depth: 3
    fig_caption: yes

fontsize: 14pt
---

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE, cache=TRUE)
```

```{r init, warning=FALSE, message=FALSE}
setwd("/Users/pedro/Documents/GitHub/Cotton-Soil-Micro/")
set.seed("12345678")
library(dada2);packageVersion("dada2") # 1.30.0
library(phyloseq);packageVersion("phyloseq") # 1.46.0
library(Biostrings); packageVersion("Biostrings") # 2.70.3
library(rmarkdown); packageVersion("rmarkdown") # 2.26
library(tidyverse); packageVersion("tidyverse") # 2.0.0
```

```{r more preparation, warning=FALSE, message=FALSE}
path.out <- "Results/Figures/"
path.rds <- "Input/RDS/"
theme_set(theme_bw())
```

Load Data (quality-trimmed and denoised using DADA2 and the nf-core/ampliseq pipeline)

```{r phyloseq-load-improved-data, eval=FALSE}
p_data.chimera_free <- readRDS("../Input/RDS/DADA2_table.rds")
```

```{r assign taxonomy, eval=FALSE}
# Assign taxonomy
tax <- assignTaxonomy(p_data.chimera_free, "../Input/rdp_19_toGenus_trainset.fa.gz", minBoot = 80, tryRC = TRUE, multithread=TRUE) # Slowest part
head(unname(tax))
tax <- addSpecies(tax, "../Input/rdp_species_assignment_18.fa.gz")
```

```{r save-progress-taxonomy, eval=FALSE}
# Write to disk
saveRDS(tax, "../Input/RDS/tax_soilmicrobes.rds")
```

Make Phyloseq object

```{r phyloseq-load-data, eval=FALSE}
p_data.chimera_free <- readRDS("../Input/RDS/DADA2_table.rds")
tax <- readRDS("../Input/RDS/tax_soilmicrobes.rds")
metadata <- read.table("../Input/metadata_matthew.txt", sep = "\t", header = TRUE)
samples.out <- data.frame(old_ID = rownames(p_data.chimera_free))
samples_out <- samples.out %>% separate_wider_delim(old_ID,delim ="_1.f", names = c("ID", "suffix"), cols_remove = FALSE)
metadata_merge <- left_join(metadata, samples_out, by = "ID")
row.names(metadata_merge) <- metadata_merge[,6]
```

```{r backup-data, eval=FALSE}
write_tsv(metadata_merge, "../Input/metadataPhyloseqFormatted.tsv")
```


```{r phyloseq-make-object-1, eval=FALSE}
p_data <- phyloseq(otu_table(p_data.chimera_free, taxa_are_rows = FALSE),
                   sample_data(metadata_merge),
                   tax_table(tax))
```

Store DNA sequences in a different slot and simplify ASV names

```{r phyloseq-format-data, eval=FALSE}
dna <- Biostrings::DNAStringSet(taxa_names(p_data))
names(dna) <- taxa_names(p_data)
p_data <- merge_phyloseq(p_data, dna)
taxa_names(p_data) <- paste0("ASV", seq(ntaxa(p_data)))
p_data
```

Investigate taxa in the dataset - how many samples are not bacteria?

```{r summarize-no-bacteria, eval=FALSE}
p_data
p_data %>%
  subset_taxa(Kingdom!="Bacteria") %>%
  tax_table() %>% as.data.frame() %>% group_by(Class) %>% tally()
```

One Plant taxa, and thirty-two Archaea.

How many identified and unidentified ASV at the Phylum and Class levels?

```{r summarize-no-bacteria-phylum-class, eval=FALSE}
p_data
p_data %>%
  tax_table() %>% as.data.frame() %>% group_by(Phylum, Class) %>% tally() %>% paged_table()
```

A few NA. Let's exclude obvious contaminants before saving sequences as a fasta file and using Blast to (hopefully) get higher definition for taxa currently classified as "NA".

First, I will make data easier to work with, by transforming the phyloseq object into a summary table.

```{r relative-abundance-dada2-processed-data-and-taxonomy, eval=FALSE}
pdata_df <- p_data %>% psmelt() %>% arrange(OTU) %>% dplyr::rename(ASV = OTU) %>% spread(Sample, Abundance)
```

Let's remove chloroplasts, plant DNA and other taxa that may not be relevant here.

```{r remove-chloroplast--plants-archaea, eval=FALSE}
p_data_taxlean <- p_data %>% subset_taxa(Kingdom != "Archaea") %>% subset_taxa((Order != "Chloroplast") | is.na(Order)) %>% subset_taxa((Kingdom != "Eukaryota") | is.na(Kingdom)) %>% subset_taxa((Class != "Cyanobacteria") | is.na(Order)) %>% subset_samples(sample_sums(p_data) > 0)
```

Save taxonomy table and ASV fasta and blast unknown sequences to increase accuracy of classification

```{r save-taxonomy-table, eval=FALSE}
p_data_taxlean %>%
  tax_table() %>%
  write.csv("../Results/MG_asv_04_2025.csv")
```

```{r save-sequences-fasta, eval=FALSE}
p_data_taxlean %>%
      refseq() %>%
      Biostrings::writeXStringSet("../Results/MG_asv_04_2025.fasta", append=FALSE,
                                  compress=FALSE, compression_level=NA, format="fasta")
```

Blast was ran using the script "blast.sh", and taxonomical ranks were added to the output file ("28APR2025_nt_blast_MG.tsv") using the script "Get_lineage_blast.sh". The resulting spreadsheet was saved as "soil_16S_withtaxranks_ASV_nt_blast.csv". Next, I will identify "NA" sequences that were resolved by blast, and manually inspect their identification and eliminate any spurious results (e.g. annotations containing mistakes, non-taxonomical attributions, etc).

```{r load taxonomical data}
old_tax_table <- read.csv("../Results/MG_asv_04_2025.csv")
blast_tax_table <- read.csv("../Results/blast_results/soil_16S_withtaxranks_ASV_nt_blast.csv")
NA_old_tax_table <- old_tax_table %>% filter(is.na(Genus)) %>% 
  rename("qseqid" = "X") #select only ASV with NA at the Genus level; rename ASV id column to match the corresponding blast column name
NoNA_blast_tax_table <- blast_tax_table %>% filter(!is.na(genus)) # select blast results that resolved identification at the Genus level
resolved_by_blast <- left_join(NA_old_tax_table, NoNA_blast_tax_table, by = "qseqid") %>% filter(!is.na(genus)) # combine both subset data sets, using the old tax table as the reference for rows to include, and qseqid as the column in common to allow the merge of these tables; next, exclude "genus" in blast results that were not resolved
```

Blast helped resolve the identification of 214 ASVs. Next, I will transfer all taxonomical levels for resolved genera from blast results to the "old_tax_table".

```{r update taxonomy}
# Prepare resolved_by_blast to be added to the final taxonomy file
# Select only relevant columns
resolved_by_blast_slim <- resolved_by_blast %>% select(c("qseqid", "kingdom", "phylum", "class", "order", "family", "genus", "species"))
# Rename columns to match phyloseq tax table
colnames(resolved_by_blast_slim) <- c("X", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
# Rename values in "Species" to eliminate characters that Phyloseq might not like
resolved_by_blast_slim <- resolved_by_blast_slim %>% mutate(Species = paste ("sp"))
# Use anti-join to eliminate from old_tax_table entries that overlap with the new results
old_tax_table_slice <- anti_join(old_tax_table, resolved_by_blast_slim, by = "X")
# Now join rows from both datasets into the final dataset
updated_tax <- bind_rows(resolved_by_blast_slim, old_tax_table_slice)
```

For now this should be sufficient. Downstream, I will further improve taxonomy by replacing NA at the genus level with the next best taxonomical id followed by "sp" (e.g. "Pseudomonadaceae sp.", or "Bacteria sp.", etc).

Add new taxonomy back to the phyloseq object

```{r add-updated-taxonomy}
#rename rows with ASV identifier
row.names(updated_tax) <- updated_tax$X
# remove first column
updated_tax <- updated_tax[-1]
new_tax <- tax_table(as.matrix(updated_tax))
tax_table(p_data_taxlean) <- new_tax
```

```{r what-taxa-is-there-2, eval=FALSE}
updated_tax %>% select(Kingdom) %>% unique()
updated_tax %>% select(Phylum) %>% unique()
updated_tax %>% select(Class) %>% unique()
updated_tax %>% select(Order) %>% unique()
```

Some taxa are classified as "bacteria" other have a different name for their respective Kingdom rank (but are still in the Bacteria domain). To make things consistent, we will rename all Kingdom as "Bacteria".

```{r rename-Kingdom}
new_tax2$Kingdom <- ifelse(new_tax2$Kingdom != "Bacteria", "Bacteria", new_tax2$Kingdom)
new_tax2 <- tax_table(as.matrix(new_tax2))
tax_table(p_data_taxlean) <- new_tax2
```

```{r save-updated_taxonomy, eval=FALSE}
saveRDS(p_data_taxlean,"../Input/RDS/p_data_taxlean.rds")
```

```{r save-ASV-metadata-seq-table, eval=FALSE}
write.table(p_data_taxlean %>% psmelt() %>% arrange(OTU) %>% 
              dplyr::rename(ASV = OTU) %>% select(ASV, Kingdom, Phylum, Class, Order, Family, Genus, Sample, Abundance) %>% spread (Sample, Abundance), 
    file = "../Results/soilmicrobes_taxclean.tsv", sep = "\t", quote = F, row.names = F, col.names = T)
```


## Decontam
```{r decontam-eval-data}
library(decontam)
df <- as.data.frame(sample_data(p_data_taxlean))
df$LibrarySize <- sample_sums(p_data_taxlean)
df <- df[order(df$LibrarySize),]
df$Index <- seq(nrow(df))
ggplot(data=df, aes(x=Index, y=LibrarySize, color=Sample_type)) + geom_point()
sample_data(p_data_taxlean)$is.neg <- sample_data(p_data_taxlean)$Sample_type == "control"

# adopting prevalence method to identify contaminants based on controls
# sensitivity set to low (threshold = 0.1)
contamdf.prev.01 <- isContaminant(p_data_taxlean, method="prevalence", neg="is.neg")
table(contamdf.prev.01$contaminant)
contaminant.list <- which(contamdf.prev.01$contaminant)
tax_table(p_data_taxlean)[contaminant.list,]

# sensitivity set to high (threshold = 0.5)
contamdf.prev.05 <- isContaminant(p_data_taxlean, method="prevalence", neg="is.neg", threshold = 0.5)
table(contamdf.prev.05$contaminant)
contaminant.list <- which(contamdf.prev.05$contaminant)
tax_table(p_data_taxlean)[contaminant.list,]
```

We identified via PCR that some reagents (e.g. some primer-barcode combinations) were likely contaminated with plant DNA.
Suspect contaminants such as Citrus only show up as contaminant when sensitivity is increased to 0.5.

Let's inspect how contaminants prevalence is divided between control and samples

```{r decontam-eval-distribution}
# Make phyloseq object of presence-absence in negative controls and true samples
ps.pa <- transform_sample_counts(p_data_taxlean, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$Sample_type == "control", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$Sample_type == "sample", ps.pa)
# Make data.frame of prevalence in positive and negative samples
# for threshold=0.1
df.pa1 <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                      contaminant=contamdf.prev.01$contaminant)
ggplot(data=df.pa1, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
  xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)") + ggtitle("Decontamination threshold=0.1")
# for threshold=0.5
df.pa5 <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                      contaminant=contamdf.prev.05$contaminant)
ggplot(data=df.pa5, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
  xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)") + ggtitle("Decontamination threshold=0.5")

p_data.ord <- ordinate(p_data, "PCoA", "bray")
plot_ordination(p_data, p_data.ord, type="sample", color="Sample_type", shape="Irrigation")+ geom_point(size=3)
```

Brevudimonas and Achromobacter are likely true members of the soil microbiome. Their ASV number also indicates that they are abundant in samples, but not as abundant in the controls. ASV2617 (Pseudomonas), on the other hand, seems to be generally less abundant overall and, perhaps, a contaminant coming from reagents or the environment during extractions. To be conservative, we will keep Brevudimonas and Achromobacter and only exclude the more clear candidate contaminant, ASV 2617.


```{r decontam-clean-data}
TaxaRemove = "ASV2617"
TaxaKeep <- setdiff(taxa_names(p_data_taxlean), TaxaRemove)
p_data.clean <- prune_taxa(TaxaKeep, p_data_taxlean)
p_data_taxlean
p_data.clean

# Remove control samples from analysis
p_data.clean <-subset_samples(p_data.clean, Sample_type!="control")
# Remove samples with zero ASV (zero column sums)
p_data.clean <- subset_samples(p_data.clean, sample_sums(p_data.clean) > 0)
```


Save progress
```{r save-decontam, eval=FALSE}
saveRDS(p_data.clean,"../Input/RDS/p_data.clean.rds")
```

```{r save-ASV-metadata-seq-table-decontam, eval=FALSE}
write.table(p_data.clean %>% psmelt() %>% arrange(OTU) %>% 
              dplyr::rename(ASV = OTU) %>% select(ASV, Kingdom, Phylum, Class, Order, Family, Genus, Sample, Abundance) %>% spread (Sample, Abundance), 
    file = "../Results/soilmicrobes_after_decontam.tsv", sep = "\t", quote = F, row.names = F, col.names = T)
```

----------------
#Stats and Plots
----------------

```{r load-more-packages}
library(microbiome)
library(ggpubr)
library(knitr)
library(ggtext)
library(scales)
library(ggside)
library(RColorBrewer)
#devtools::install_github("vmikk/metagMisc")
library(metagMisc)
```

First step is to rarefy data to account for differences in sequencing depth by sample.

For rarefying data, I will use phyloseq_mult_rare_avg, that will subsample samples 1000 times and average the number of times ASVs are observed in each sample according to the threshold I will establish below

```{r thresholds-rarefy-data, eval=FALSE}
sample_size_quantile <- as.data.frame(quantile(sample_sums(p_data.clean), probs = c(0, 0.10, 0.15, 0.225, 0.25, 0.5, 0.75, 1)))
sample_size_quantile["10%", ] # 12321.6 reads
sample_size_quantile["25%", ] # 13716.25 reads
sample_size_quantile["50%", ] # 16868.5 reads
sample_size_quantile["75%", ] # 19450.75 reads
```

Samples that would be lost with some of these thresholds

```{r samples-lost}
sample_loss <- as.data.frame(sample_sums(p_data.clean))
names(sample_loss) <- "Sample_size"
data.frame(Quantile_cutoff = c("no cutoff", "10%", "25%", "50%", "75%"), Samples_left = c(nrow(sample_loss), nrow(filter(sample_loss, Sample_size > 12321.6)), nrow(filter(sample_loss, Sample_size > 13716.25)), nrow(filter(sample_loss, Sample_size > 16868.5)), nrow(filter(sample_loss, Sample_size > 19450.75))))
```
What two samples do we lose if we were to use 10% as our cutoff?

```{r ten-percent-cutoff}
sample_loss %>% filter(Sample_size < 12321.6) #10 percent
sample_loss %>% filter(Sample_size < 13716.25) #25 percent
```

Both two samples lost if threshold is ten percent were from the same treatment (reduced drought). However one of them is closer to the cutoff than the other. We will choose 11940 as our cutoff, since lowering the cutoff to include sample gillis_18 would still keep enough reads/sample after rarefaction.

```{r rarefy-samples}
pdata_rr <- phyloseq_mult_raref_avg(p_data.clean, SampSize = 11940, iter=1000, parallel = TRUE,verbose = TRUE)

pdata_rr <- transform_sample_counts(pdata_rr, function(x){ x * 11940})
```


The phyloseq_mult_raref_avg function removes the "Refseq" component of the phyloseq object. I will put it back to the phyloseq object, so that it can be used in downstream analysis

```{r add-back-sequences}
pdata_table <- t(as(otu_table(pdata_rr), "matrix"))
asv_ids <- colnames(pdata_table)
refseq_phyloseq <- refseq(p_data.clean)
refseq_phyloseq_subset <- refseq_phyloseq[names(refseq_phyloseq) %in% asv_ids]
pdata_rr_up <- merge_phyloseq(pdata_rr, refseq_phyloseq_subset)
```


Next, let's test our hypotheses
(1) That microbiome responds to soil moisture, and we expect to observe different communities
  a. specifically we would expect to find more plant growth promoting bacterial taxa in soils with high moisture as opposed to soils exposed to drought
(2) We would also expect microbiomes to respond to soil management, although Matthew has found no effect for nematodes sampled in the same fields

In addition, we expect that sampling site region (e.g. county) will not influence microbiota composition.

To test these hypotheses, we will use a Permanova test. One assumption of Permanova is that there is homogeneity of dispersion between groups.

For hypothesis "1", "Irrigation" is the grouping variable, and for the second hypothesis "Management" is the group variable. In all cases, "County" is a blocking variable.


```{r betadisper}
library("vegan")
# Extract OTU table from phyloseq object, calculate Bray-Curtis distances, and calculate beta dispersion
otu_table <- t(as(otu_table(pdata_rr_up), "matrix"))
bray_dist <- vegdist(otu_table, method = "bray")
metadata_rr_up <- data.frame(sample_data(pdata_rr_up))
disper_Irrigation <- betadisper(bray_dist, metadata_rr_up$Irrigation)

# Test if dispersion between groups is significant
permutest(disper_Irrigation) # P-value 0.398, data variance is homogeneous among groups being tested
```

Homogeneity of dispersion is met and, therefore, we can proceed with Bray-Curtis distances for Permanova.

Test the hypothesis that microbiome is different between Irrigation treatments

```{r permanova-sp}
Irrigation <- sample_data(pdata_rr_up)$Irrigation
Tillage <- sample_data(pdata_rr_up)$Tillage
County <- sample_data(pdata_rr_up)$County
metadata <- data.frame(sample_data(pdata_rr_up))
adonis2(formula = bray_dist ~ Irrigation*Tillage, data = metadata, strata = County, permutations = 999)
```

We did not find an effect of Irrigation neither Tillage. Bray-Curtis is abundance weighted distance. Let's look at a distance that compare groups only based on presence/absence of taxa, to check if the results would still be non-significant.


```{r permanova-jaccard}
jaccard_dist <- vegdist(otu_table, method = "jaccard")
adonis2(formula = jaccard_dist ~ Irrigation*Tillage, data = metadata, strata = County, permutations = 999)
```

The results above are consistent and indicate that indeed there are no obvious difference between soil microbiomes sampled in irrigated/non-irrigated fields, neither a combination of these with reduced or conventional tillage.


************
# Figures
************

Alpha diversity

```{r combine-bac-by-family, eval=FALSE}
pseq_alpha <-aggregate_taxa(pdata_rr_up, level = "Genus")
tab <-microbiome::diversity(pseq_alpha, index = "Shannon")
```

Extract and prepare dataframe for plotting alpha diversity as a violin plot
```{r plot-violin-plots}
pseq_alpha_meta <- microbiome::meta(pseq_alpha)
pseq_alpha_meta$Shannon <- tab$shannon

pseq_alpha_plot <- ggviolin(pseq_alpha_meta, x="Irrigation", y="Shannon", add="boxplot", fill="Irrigation")

p1 <- pseq_alpha_plot +
  theme(
    axis.title.x = element_blank(),
    axis.text.x.bottom = element_blank(),
    axis.title.y = element_markdown(), 
    legend.text = element_markdown(size = 14),
    legend.title = element_markdown(size = 14))
ggpar(p1, ylim = c(0, 3))
```

```{r save-alpha-diversity-plot, eval=FALSE}
ggsave("../Results/Figures/alphadiversity.pdf", width=15, height=10)
```

PCoA plots

```{r pcoa}
# PCoA with density curves on the x and y-axes

ps_object <- pdata_rr_up
pcoa <- cmdscale(as.matrix(bray_dist), eig = TRUE, k = 2)
scores <- as.data.frame(pcoa$points)
colnames(scores) <- c("MDS1", "MDS2")

# Add metadata
metadata <- data.frame(sample_data(ps_object))
plot_data <- cbind(scores, metadata)

# Create variance variable for the plot axis labels
var_explained <- round(100 * pcoa$eig / sum(pcoa$eig), 2)

# Generate figure (PCoA and boxplots along both axis)
p1 <- ggplot(plot_data, aes(x = MDS1, y = MDS2, color = Irrigation, shape = Tillage)) +
  geom_point(size = 8, alpha = 0.5) +
  scale_color_brewer(palette = "Set2") +
  labs(
    x = paste0("MDS1 (", var_explained[1], "%)"),
    y = paste0("MDS2 (", var_explained[2], "%)"),
    color = "Irrigation",
    shape = "Tillage",
    title = "Irrigation does not affect the microbiome",
    caption = "<i>F<sub>2</sub></i>=1.0467, df=1,18, p-value=<b>0.249</b>"
  ) +
  theme_minimal() + theme(legend.text = element_text(size = 14), legend.title = element_text(size = 14), plot.title = element_markdown(), 
                          plot.caption = element_markdown()) +
  scale_shape_manual(values= 15:17) +
  ggside::geom_xsideboxplot(aes(fill = Irrigation, y = Irrigation, group = Irrigation), orientation = "y", show.legend = FALSE) +
  ggside::geom_ysideboxplot(aes(fill = Irrigation, x = Irrigation, group = Irrigation), orientation = "x", show.legend = FALSE) +
  ggside::scale_xsidey_discrete(labels = NULL) +
  ggside::scale_ysidex_discrete(labels = NULL) +
  ggside::theme_ggside_void()+
  scale_fill_brewer(palette = "Set2")


ggsave("../Results/Figures/PCoA.pdf", height = 12, width = 18)
```


```{r barplot}

# 1. Transform phyloseq object into a data frame
pseq_barplot <- pseq_alpha %>% psmelt() %>% arrange(OTU) %>% rename(ASV = OTU) %>%
        select(ASV, Kingdom, Phylum, Class, Order, Family, Genus, Sample, Abundance) %>% spread(Sample, Abundance)

# 2. Taxa are rows. Create a column for summing all reads per taxa, and order taxa so that the most abundant taxa is listed first, then the second, third, etc. Next, create a column with the cumulative sums (i.e. row 1a = row1, row2a = row1 + row2, row3a = row1 + row2 + row3)
pseq_barplot <- pseq_barplot %>% filter(!Genus == "Unknown")
pseq_barplot$abundance_sum <- rowSums(pseq_barplot[8:26])
pseq_barplot <- pseq_barplot[order(-pseq_barplot$abundance_sum),]
pseq_barplot <- pseq_barplot %>% mutate(cumulative_sums = cumsum(abundance_sum))

# 3. Determine the threshold where cumulative abundance reaches 95%
threshold <- sum(pseq_barplot$abundance_sum) * 0.95 #there are many (hundreds?) within the top most abundant taxa, even when we increase this cutoff to 99%

# 4. Because all taxa in these samples seem to differ very little in abundance (somewhat sharing eveness), then we will plot a barplot with the top 10 taxa only

pseq_barplot_slice <- pseq_barplot %>% slice_head(n=10)

# Join the pseq_barplot and metadata
data_combined <- pseq_barplot_slice[7:26] %>%
  gather(sample, count, -Genus) %>%
  dplyr::left_join(metadata, by = c("sample" = "old_ID"))

# For plotting purposes, inspect number of taxa present and order to be presented
#unique(data_combined$Genus)
#data_combined$Genus <- factor(data_combined$Genus, levels = c("Wolbachia", "Holzapfelia", "Gilliamella", "Unknown", "other"))

# Create the side-by-side bar plots using ggplot2 and facet_wrap
ggplot(data_combined, aes(x = sample, y = count, fill = Genus)) +
  geom_bar(stat = "identity", position = "fill", width = 0.9) +
  facet_grid(cols = vars(Irrigation), scale="free") + 
  labs(x = "Samples",
       y = "Relative Frequency") +
  theme(legend.position = "top", 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank())
#,strip.text = element_text(face = "italic", size = 14), legend.text = element_markdown(size = 14), legend.title = element_markdown(size = 14)) +
#  scale_fill_manual(values = c("Wolbachia" = "#009E73", "Holzapfelia" = "black", "Gilliamella" = "#CC79A7", "Unknown" = "lightgrey", "other" = "darkgrey"),
#  labels = c("<i>Wolbachia</i>","<i>Holzapfelia</i>","<i>Gilliamella</i>","Unknown","Other"))  # Customize fill colors
```

```{r savebarplot}
ggsave("../Results/Figures/barplot.pdf", width=15, height=10)
```
